code/modules/reagents/reagent_containers/glass.dm

```diff
@@ -330,3 +330,58 @@
 
 /obj/item/reagent_containers/glass/beaker/waterbottle/large/empty
 	list_reagents = list()
+
+/obj/item/pestle
+	name = "pestle"
+	desc = "i exist"
+	icon = 'icons/obj/drinks.dmi'
+	icon_state = "largebottle"
+
+/obj/item/reagent_containers/glass/mortar
+	name = "mortar"
+	desc = "Use this with a pestle to extract chemicals out of items."
+	amount_per_transfer_from_this = 10
+	possible_transfer_amounts = list(5, 10, 15, 20, 25, 30, 50)
+	volume = 100
+	reagent_flags = OPENCONTAINER
+	spillable = TRUE
+	icon = 'icons/obj/drinks.dmi'
+	icon_state = "smallbottle"
+	var/obj/item/grinded
+
+/obj/item/reagent_containers/glass/mortar/AltClick(mob/user)
+	if(grinded)
+		grinded.forceMove(drop_location())
+		grinded = null
+		to_chat(user, "You eject the item inside.")
+
+/obj/item/reagent_containers/glass/mortar/attackby(obj/item/I, mob/living/carbon/human/user)
+	if(istype(I,/obj/item/pestle))
+		if(grinded)
+			to_chat(user, "You start grinding...")
+			if(do_after(user, 25, target = src))
+				user.adjustStaminaLoss(40)

```

Comment:
Just to be sure, if you spamclick (Or lag and click multiple times) after 25 (seconds?) you may knock yourself out?

Reply:
2.5 seconds but yes

Reply:
solution:
1. make this nest less (not really part of the solution i just hate nesting)
2. check if(grinded) again